sdk/Lusid.Sdk.Tests/Features/FeatureExtractor.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Lusid.Sdk.Tests.Features
+{
+
+    public static class FeatureExtractor
+    {
+        private static List<String> GetAttributesFromClassMethods(Type classType)
+        {
+            LusidFeature att;
+            List<String> classAttributes = new List<string>();
+ 
+            MethodInfo[] myMemberInfo = classType.GetMethods();
+
+            // Loop through all methods in this class that are in the
+            // MyMemberInfo array.
+            foreach (var method in myMemberInfo)
+            {
+                att = (LusidFeature) Attribute.GetCustomAttribute(method, typeof (LusidFeature));
+                
+                if (att == null) continue;
+
+                classAttributes.Add(att.Code);
+            }
+            return classAttributes;
+        }
+        
+        public static List<String> GetAllMethodAttributesInNamespace(string nameSpace)
+        {
+            Type[] asmTypes = Assembly.GetExecutingAssembly().GetTypes();
```

Comment:
you can simplify this with the following which will give you a list of all the features

```
var features = Assembly
                .GetAssembly(typeof(FeatureExtractorTests))
                ?.GetTypes()
                .SelectMany(t => t.GetMethods()
                    .Where(m => m.GetCustomAttributes(typeof(LusidFeature), true)?.Length > 0)
                    .Select(m => m.GetCustomAttributes(typeof(LusidFeature)).Cast<LusidFeature>().First().Code)
                );
```

1.`Assembly.GetAssembly(typeof(FeatureExtractorTests))` gets the Assembly for the type `FeatureExtractorTests`
2. Get all the types
3. `SelectMany` will combine the results for each `Type` into a single list
4. For each `Type`, get the methods and filter where it has a `LusidFeature` attribute then select the `First()` LusidFeature` attribute and extract the code
5. What you get back is an `IEnumerable<string>` which is the list of features

Reply:
I love this. I also added a part where I'm only selected from a specified namespace, and I'm allowing the namespace to be case insensitive - the rationale for that is because I've noticed that the 'Properties' features test is under '..tutorials' namespace with lowercase, and others are under '...Tutorials'. Also collecting the resulting stream into a list

`            var featureList = Assembly
                .GetExecutingAssembly()
                .GetTypes()
                .Where(t => t.Namespace != null && t.Namespace.ToLower().StartsWith(nameSpace.ToLower()))
                .SelectMany(t => t.GetMethods()
                    .Where(m => m.GetCustomAttributes(typeof(LusidFeature), true)?.Length > 0)
                    .Select(m => m.GetCustomAttributes(typeof(LusidFeature)).Cast<LusidFeature>().First().Code)
                ).ToList();`

Reply:
c# is case sensitive so you don't need to worry about the namespacing vs. folder (vs python which uses folders)

Reply:
Hmm that's strange, because if you look at the c# sdk and go into tutorials -> ibor, and if you compare 'Properties.cs' versus eg. 'Holdings.cs', you will see that the namespace for Properties is **_Lusid.Sdk.Tests.Tutorials.Ibor_** and for Holdings it's **_Lusid.Sdk.Tests.Tutorials.Ibor_**. When I don't make the namespace search case insensitive, I don't get back the features in 'Properties.cs' file.

Reply:
What I meant was you don't need to filter on the namespace, you just need to search the assembly for any class that has the `LusidFeature` attribute

Reply:
I don't think it's desirable to get the features by manually entering class names, as every time a new class would be added, we would need to add that into the test cases. That's why it's better to filter by namespace. 

In addition to that, we don't want to scan through the entire project for all methods with LusidFeature attribute, because it will also include the LusidFeature attributes that we want to be isolated for unit tests only

Reply:
good point bit the tests, you're right  you probably do need to filter them out